edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/authenticators/DeviceScopeCertificateAuthenticator.cs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.CloudProxy.Authenticators
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Security.Cryptography;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Common.Data;
+    using Microsoft.Azure.Devices.Common.Security;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Device;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity.Service;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class DeviceScopeCertificateAuthenticator : IAuthenticator
```

Comment:
Sorry, forgot to mention earlier.. we should derive this from DeviceScopeAuthenticator.
(I am happy to refactor this later if you feel it is too late right now).

Reply:
Fixed.